android/quest/src/main/java/org/smartregister/fhircore/quest/ui/task/PatientTaskFragment.kt

```diff
@@ -0,0 +1,111 @@
+ */
+
+package org.smartregister.fhircore.quest.ui.task
+
+import android.os.Bundle
+import android.view.View
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.lifecycle.ViewModelProvider
+import androidx.paging.compose.LazyPagingItems
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import org.smartregister.fhircore.engine.configuration.view.RegisterViewConfiguration
+import org.smartregister.fhircore.engine.ui.register.ComposeRegisterFragment
+import org.smartregister.fhircore.engine.ui.register.RegisterDataViewModel
+import org.smartregister.fhircore.engine.ui.register.model.RegisterFilterType
+import org.smartregister.fhircore.engine.util.ListenerIntent
+import org.smartregister.fhircore.engine.util.extension.createFactory
+import org.smartregister.fhircore.quest.data.task.PatientTaskRepository
+import org.smartregister.fhircore.quest.data.task.model.PatientTaskItem
+import org.smartregister.fhircore.quest.ui.task.component.PatientTaskList
+import org.smartregister.fhircore.quest.util.QuestConfigClassification
+
+@AndroidEntryPoint
+class PatientTaskFragment : ComposeRegisterFragment<PatientTask, PatientTaskItem>() {
+
+  @Inject lateinit var patientTaskRepository: PatientTaskRepository
+
+  override lateinit var registerViewConfiguration: RegisterViewConfiguration
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    registerViewConfiguration =
+      configurationRegistry.retrieveConfiguration<RegisterViewConfiguration>(
+        configClassification = QuestConfigClassification.PATIENT_TASK_REGISTER,
+      )
+    configureViews(registerViewConfiguration)
+  }
+
+  override fun navigateToDetails(uniqueIdentifier: String) {
+    // TODO: Nothing to navigate at the moment
+  }
+
+  @Composable
+  override fun ConstructRegisterList(
+    pagingItems: LazyPagingItems<PatientTaskItem>,
+    modifier: Modifier
+  ) {
+    PatientTaskList(
+      pagingItems = pagingItems,
+      useLabel = registerViewConfiguration.useLabel,
+      modifier = Modifier,
+      clickListener = { listenerIntent, data -> onItemClicked(listenerIntent, data) }
+    )
+  }
+
+  override fun onItemClicked(listenerIntent: ListenerIntent, data: PatientTaskItem) {
+    // Nothing to navigate at the moment
+  }
+
+  // todo maybe we need to do a db call
+  override fun performFilter(
+    registerFilterType: RegisterFilterType,
+    data: PatientTaskItem,
+    value: Any
+  ): Boolean {
+    return when (registerFilterType) {
+      RegisterFilterType.SEARCH_FILTER -> {
+        if (value is String && value.isEmpty()) return true
+        else
+          data.name.contains(value.toString(), ignoreCase = true) ||
+            data.id.contentEquals(value.toString())
+      }
+      RegisterFilterType.OVERDUE_FILTER -> {
+        data.overdue
+      }
+    }
+  }
+
+  @Suppress("UNCHECKED_CAST")
```

Comment:
can we check the type before we cast and remove this suppression?

Reply:
Yeah, I think we can do something similar